### PR TITLE
retrofit v4

### PR DESCRIPTION
**Pull request by nbhd**
https://github.com/nbhd/welcome-to-android/issues/19
- [x] @shaunkawano 
- [x] @eoinfogarty 
- [x] @bowyer-app

### CODE REVIEW

File: grepos/app/src/main/java/net/boondockradio/grepos/MainActivity.java [hunk 1]
```diff
@@ -93,28 +93,21 @@ private void fetchRepositories() {
...
-                        mProgress.setVisibility(View.GONE);
-
-                        mAdapter.add(repositories.items);
-                        mAdapter.setIsLoading(false);
-                        isLoading = false;
-                        mTotal = mAdapter.getItemCount();
-                    }
-                });
+                .subscribe(
+                        repositories -> {
```
Comment (shoheikawano) [inline grepos/app/src/main/java/net/boondockradio/grepos/MainActivity.java]: 👍

Comment (eoinfogarty) [inline grepos/app/src/main/java/net/boondockradio/grepos/MainActivity.java]: good 👍 

just for fun can you make it like this?
`.subscribe(this::handleResponse, this::handleError, this::handleComplete)`

Comment (nbhd) [inline grepos/app/src/main/java/net/boondockradio/grepos/MainActivity.java]: sure 🌵

File: grepos/app/src/main/java/net/boondockradio/grepos/MainActivity.java [hunk 2]
```diff
@@ -93,28 +93,29 @@ private void fetchRepositories() {
...
-                        mProgress.setVisibility(View.GONE);
-
-                        mAdapter.add(repositories.items);
-                        mAdapter.setIsLoading(false);
-                        isLoading = false;
-                        mTotal = mAdapter.getItemCount();
-                    }
-                });
+                .subscribe(
+                        this::handleResponse,
```
Comment (shoheikawano) [inline grepos/app/src/main/java/net/boondockradio/grepos/MainActivity.java:97]: nice! it's very easy to read.

Comment (nbhd) [inline grepos/app/src/main/java/net/boondockradio/grepos/MainActivity.java:97]: ＿人人人人人人人人人人人＿
＞　easy to read lambda　＜
￣Y^Y^Y^Y^Y^Y^Y^Y^Y^Y￣

File: grepos/build.gradle [hunk 3]
```diff
@@ -6,6 +6,7 @@ buildscript {
     }
     dependencies {
         classpath 'com.android.tools.build:gradle:2.1.2'
+        classpath 'me.tatarka:gradle-retrolambda:3.2.5'
```
Comment (shoheikawano) [inline grepos/build.gradle:9]: This is just an additional info; currently, this gradle-retrolambda library itself uses old version of retrolambda by default, so if we were to reference to the latest retrolambda, we need to `configure` it. 
https://github.com/evant/gradle-retrolambda#using-a-different-version-of-the-retrolambdajar

**Comment by bowyer-app** [comment]
LGTM

File: grepos/build.gradle (hunk 3 above)
Comment (nbhd) [inline grepos/build.gradle:9]: thanks!!

**Comment by shoheikawano** [comment]
LGTM

**Comment by eoinfogarty** [comment]
LGTM

**Pull request merged**